### ISSUE TITLE
December 2019 - content update

### ISSUE THREAD

**Issue by DawnmarieDesJardins** [microsoft/MCW-Cognitive-services-and-deep-learning#30]
This workshop is scheduled for a content update. Please review the current workshop and provide update suggestions for review. Thanks!

**Comment by shirolkar** [comment]
Following are some suggestion to update and simply the MCW.

Update deep neural network text classification model to use Glove embeddings instead of TF-ID vectorizer. Glove embeddings-based text vectorization is used in some of the new MCW’s like “Machine Learning”. This change will impact the following files:
- textanalytics.py: Simplify this required library and remove dependencies on the nltk package.
- 00 Init notebook: Remove this notebook and thus the need to download nltk components. Downloading nltk components is very time consuming.
- 03 Claim Classification notebook: Download the preprocessed Glove dictionary and word vectors. Use Keras preprocessing libraries for tokenization and padding sequences and update the neural net architecture to use Glove embedding for text vectorization. Remove download of nltk components. Update the notebook documentation.
- 04 Deploy Classifier Web Service notebook: Update model registration section, rewrite the scoring script, update notebook documentation.
- Test end to end.

Exercise 4: Completing the solution
- Update affected screenshots to match the new Azure portal UI.

Please provide your feedback. Thanks.

**Comment by soniaang** [comment]
Hi I really like  this very concise case study which makes it very simplistic, The technology is there and it mixes AML with Deep learning with Cognitive OCR and Sentiment analysis 
Some comments to make this more meaningful  to teh non insurance person
Can you put a short blurb why we are doing classification of claims and how sentiment is used for claims analysis -- in fact there must be a feedback loop from sentiment analysis in  the  process  and the arrows here I am not quite agreeing. The processes may be better represented as continued from one to the other.  

In the intro to the work shop can you add some business background like this .. just summarize this
But maybe some short blurb on why we are doing claims classification in insurance for those who are not insurance savvy?
Insurance claims handlers perform analysis to decide which claims are valid -- and eligible for payout for settlement  --and which may be fraudulent. Cases of potential fraud are referred to a special investigations unit (SIU) for further scrutiny. Suspect claims can be detected more quickly by tracking recognized suspicious indicators through first notification of loss (FNOL) systems. Enhanced billing analytics can help isolate cases of inflated claims. However, these systems are still generally limited to an individual insurance provider’s silos of data , a limitation that can make it possible for an individual to defraud multiple insurers with the same scam.
Sentiment analysis datasets can be effectively utilized by insurance companies to classify complaints according to the products, services, or operations as it eases out the claim management process for customers and companies and ensures that customers receive timely responses. Moreover, they reduce the pressure of customer service centers and help in identifying fraud and fraudulent customers. Companies can even leverage social media sentiment analysis datasets to enhance customer experience, improve improvs client recommendations and retention, gain a market overview, and reduce indirect costs.

**Comment by swan-am-i** [comment]
Would be good to explain the architectural differences between choosing TF-IDF, n-gram and word2vec and when to use a lemmatizer and use a stemmizer. Would be good to show how we should work with data sparcity. Would be good to discuss the boundaries of LUIS etc

**Comment by kayapperson** [comment]
It's quite impressive (and also a lot, so I couldn't look at everything yet). However, I really appreciate the coverage of this workshop! It includes both prebuilt AI and DIY AI. I looked at the whiteboard materials, and found they were well written.

For the HoL part, could you please add a table that specifies how long it should take for each step as a guideline? This would give both the instructor and participants a rough idea of their investment of time.  

Lastly, the presentation layer. Is it not included in the workshop? The workshop covers the core AI pieces well, and so please consider adding how all of these are pulled together as a solution. It'd be very fulfilling to have the finished solution. This could be as a separate module. A simple web app or PowerApps would do. 

Thank you for another awesome work once again!

**Comment by shirolkar** [comment]
> Hi I really like this very concise case study which makes it very simplistic, The technology is there and it mixes AML with Deep learning with Cognitive OCR and Sentiment analysis
> Some comments to make this more meaningful to teh non insurance person
> Can you put a short blurb why we are doing classification of claims and how sentiment is used for claims analysis -- in fact there must be a feedback loop from sentiment analysis in the process and the arrows here I am not quite agreeing. The processes may be better represented as continued from one to the other.
> 
> In the intro to the work shop can you add some business background like this .. just summarize this
> But maybe some short blurb on why we are doing claims classification in insurance for those who are not insurance savvy?
> Insurance claims handlers perform analysis to decide which claims are valid -- and eligible for payout for settlement --and which may be fraudulent. Cases of potential fraud are referred to a special investigations unit (SIU) for further scrutiny. Suspect claims can be detected more quickly by tracking recognized suspicious indicators through first notification of loss (FNOL) systems. Enhanced billing analytics can help isolate cases of inflated claims. However, these systems are still generally limited to an individual insurance provider’s silos of data , a limitation that can make it possible for an individual to defraud multiple insurers with the same scam.
> Sentiment analysis datasets can be effectively utilized by insurance companies to classify complaints according to the products, services, or operations as it eases out the claim management process for customers and companies and ensures that customers receive timely responses. Moreover, they reduce the pressure of customer service centers and help in identifying fraud and fraudulent customers. Companies can even leverage social media sentiment analysis datasets to enhance customer experience, improve improvs client recommendations and retention, gain a market overview, and reduce indirect costs.

Hello Sonia Ang:

Thank you for your feedback. The customer case study described in the MCW talks about the needs for classification and sentiment analysis to address the following use cases:  

When an agent (an employee or contractor of Contoso) is processing a claim, there are multiple challenges that add significantly to the cost, including the significant time it takes for an agent to read through and process the content submitted with each claim, as well as the difficulty they have in finding particular claim artifacts when returning to a claim after a while. While each claim is stored in a database, the details about the claim, including the free text responses and supporting photos, are stored as opaque attachments that are not searchable - meaning agents typically pull up the claim by the claim number or the insured's contact information and then must manually read through the attachments.

One set of such issues deals with the free text responses. The first issue Contoso identified is that each claim detail should be automatically classified as either home or auto based on the text. This classification would be displayed in the claim summary, so an agent can quickly assess whether they are dealing with purely a home claim, an auto claim or a claim that has a mixture of the two.

The second issue is Contoso would like to experiment applying sentiment analysis to the claim text. They know most customers are either factual in their description (a neutral sentiment) or slightly unhappy (a more negative sentiment), but believe that a negative sentiment can be an indicator to claim text that involves a more severe situation, which might warrant an expedited review by an agent.

The third issue with the free text is that some of the responses are long. When agents are shifting between claims, it can be hard for them to recall which response had the details for which they are looking. Contoso would like to experiment with an automatic summarization of long claims that produces a summary of about 30 words in length. This summarization would enable the agent to get the gist before having to read the full claim and can quickly remind themselves of the claim when revisiting it.

The use case is described in the whiteboard design session part of the MCW. The MCW, including the hands-on lab is designed towards addressing this use case. 

Are you thinking that the abstract needs to focus more on describing the use case instead of the technology components used in the MCW? Thanks for your clarification. 

Jitendra

**Comment by shirolkar** [comment]
> It's quite impressive (and also a lot, so I couldn't look at everything yet). However, I really appreciate the coverage of this workshop! It includes both prebuilt AI and DIY AI. I looked at the whiteboard materials, and found they were well written.
> 
> For the HoL part, could you please add a table that specifies how long it should take for each step as a guideline? This would give both the instructor and participants a rough idea of their investment of time.
> 
> Lastly, the presentation layer. Is it not included in the workshop? The workshop covers the core AI pieces well, and so please consider adding how all of these are pulled together as a solution. It'd be very fulfilling to have the finished solution. This could be as a separate module. A simple web app or PowerApps would do.
> 
> Thank you for another awesome work once again!


Hi Kay Apperson:
Thank you for your positive feedback. We should be able to provide time estimates for the HoL sections. Since the focus of the MCW is on Cognitive Services and Deep Learning we did not extend the experience to include UI. It also helps to limit the overall experience to be completed within 4 hours. Since both Cognitive Services and deployed Deep Learning models are webservices, they are suitable to be consumed by an end user facing application. In the lab we have used HTML rendering in python notebook to show how different pieces of information from different services can be brought together in one output. 
Jitendra

**Comment by DawnmarieDesJardins** [comment]
@shirolkar @timahenning 
Hello!  Please see open issue #31 while working this update.
Thank you.

**Comment by shirolkar** [comment]
> Would be good to explain the architectural differences between choosing TF-IDF, n-gram and word2vec and when to use a lemmatizer and use a stemmizer. Would be good to show how we should work with data sparcity. Would be good to discuss the boundaries of LUIS etc
@DawnmarieDesJardins @swan-am-i @timahenning 
Hello Priya - Thanks for your feedback. I plan to add a new question in the white board session section that will address the key aspects of your feedback that are applicable to the claims classification part of the solution.
Regards,
Jitendra

**Comment by DawnmarieDesJardins** [comment]
Closing issue with December 2019 update merge.
